lib/syskit/roby_app/plugin.rb

```diff
@@ -146,8 +148,64 @@ def self.setup(app)
 
                 rtt_core_model = app.default_loader.task_model_from_name("RTT::TaskContext")
                 Syskit::TaskContext.define_from_orogen(rtt_core_model, register: true)
+
+                # Log Transfer FTP Server spawned during Application#setup
+                start_tmp_root_ca
+
+                start_local_log_transfer_server(
+                    app.log_dir,
+                    @tmp_root_ca.ca_user,
+                    @tmp_root_ca.ca_password,
+                    @tmp_root_ca.signed_cert_filepath
+                )
+            end
+
+            def self.send_file_transfer_command(name, tmp_root_ca, logfile)
+                # Checks if said process server exists and is connected
+                if process_servers[name]
+                    # Establishes communication with said process server
+                    client = Syskit::RobyApp::RemoteProcesses::Client.new(
+                            host: process_server[name].host_id,
+                            port: @server_port)
+                    # Commands method log_upload_file from said process server
+                    client.log_upload_file(
+                        host: "localhost", 
```

Comment:
host cannot be `localhost`. `localhost` is always the machine __of the process that tries to create a connection__

In other words, this instructs the process server to connect to ftpd server that is on its own machine. Kind of defeats the purpose ...

Add the `host` parameter as a `#public_ip` property on Plugin (which is then accessible on the app object, that is Roby.app).  Do not start the log transfer server if it is not set.